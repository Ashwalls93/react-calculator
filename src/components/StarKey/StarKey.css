.star-key-container {
  align-items: center;
  display: flex;
  justify-content: center;
  transition: var(--default-transition);
  width: 25%;
}

.star-key-container:hover {
  cursor: pointer;
}

.star-key-theme-default {
  background-color: var(--star-key-color-default);
}

.star-key-theme-default:hover {
  background-color: var(--star-key-color-default-hover);
}

.star-key-theme-pony {
  background-color: var(--star-key-color-pony);
  border-radius: 8px;
}

.star-key-theme-pony:hover {
  background-color: var(--star-key-color-pony-hover);
}

.star-key-theme-dondi {
  background-color: var(--star-key-color-dondi);
  border-radius: 8px;
  box-shadow: var(--key-border);
}

.star-key-theme-dondi:hover {
  background-color: var(--star-key-color-dondi-hover);
  /* box-shadow: var(--key-border-hover); */
}

.star-icon {
  height: var(--star-icon-size);
  transition: var(--default-transition);
  width: var(--star-icon-size);
}

.star-theme-default {
  fill: var(--star-icon-color-default);
}

.star-theme-pony {
  fill: var(--star-icon-color-pony);
}

.star-theme-dondi {
  fill: var(--star-icon-color-dondi);
}