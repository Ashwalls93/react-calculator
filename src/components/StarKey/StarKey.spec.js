import React from 'react';
import {shallow} from 'enzyme';
import StarKey from './StarKey';

describe('StarKey', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <StarKey 
        theme={'default'}
        toggleTheme={jest.fn()} 
      />
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should render an <svg />', () => {
    expect(wrapper.find('svg').length).toEqual(1);
  });

  it('should have a className reflecting its theme', () => {
    expect(wrapper.find('.star-theme-default').exists()).toBe(true);
  });

  it('should update its className on theme change', () => {
    expect(wrapper.find('.star-theme-default').exists()).toBe(true);
    wrapper.setProps({theme: 'pony'});
    expect(wrapper.find('.star-theme-pony').exists()).toBe(true);
    wrapper.setProps({theme: 'dondi'});
    expect(wrapper.find('.star-theme-dondi').exists()).toBe(true);
  });

  it('should have the correct icon path based on theme', () => {
    const themes = ['default', 'pony', 'dondi'];
    themes.forEach((theme) => {
      const testWrapper = shallow(
        <StarKey 
          theme={theme}
          toggleTheme={jest.fn()} 
        />
      );
      wrapper.setProps({theme});
      const wrapperPath = wrapper.find('path').first().html();
      const testPath = testWrapper.find('path').first().html();
      expect(wrapperPath === testPath).toEqual(true);
    });
  });
});