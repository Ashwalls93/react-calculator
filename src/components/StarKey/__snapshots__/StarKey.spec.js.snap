// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarKey should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StarKey
    theme="default"
    toggleTheme={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <svg
        className="star-icon star-theme-default"
        viewBox="0 0 24 24"
>
        <path
                d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
        />
        <path
                d="M0 0h24v24H0z"
                fill="none"
        />
</svg>,
      "className": "star-key-container star-key-theme-default",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
/>,
          <path
            d="M0 0h24v24H0z"
            fill="none"
/>,
        ],
        "className": "star-icon star-theme-default",
        "viewBox": "0 0 24 24",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "d": "M0 0h24v24H0z",
            "fill": "none",
          },
          "ref": null,
          "rendered": null,
          "type": "path",
        },
      ],
      "type": "svg",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <svg
          className="star-icon star-theme-default"
          viewBox="0 0 24 24"
>
          <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
          <path
                    d="M0 0h24v24H0z"
                    fill="none"
          />
</svg>,
        "className": "star-key-container star-key-theme-default",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <path
              d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
/>,
            <path
              d="M0 0h24v24H0z"
              fill="none"
/>,
          ],
          "className": "star-icon star-theme-default",
          "viewBox": "0 0 24 24",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "d": "M0 0h24v24H0z",
              "fill": "none",
            },
            "ref": null,
            "rendered": null,
            "type": "path",
          },
        ],
        "type": "svg",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
